packages/grafana-data/src/types/panel.ts

```diff
@@ -96,6 +96,12 @@ export interface PanelProps<T = any> {
   /** EventBus  */
   eventBus: EventBus;
 
+  /** For instance state that can be shared between panel & options UI  */
+  instanceState?: S;
```

Comment:
maybe better if we attach this to panelContext rather than passed via props.  I expect the usage to be very rare and pretty spetalized.

Similarly could/should the panelContext be passed to the options builder? rather than an explicit state (brainstorming, not sure it is a good idea!)

Reply:
I think I agree with @ryantxu on this one. I would keep panel props as clean as possible not to bring confusion to plugin authors.

Reply:
Ah damn, but then the instance state won't be available in the panel edit as there is no PanelContext wrapping entire edit container IIRC.

Reply:
maybe, it makes it a bit more complex as the context is a separate state that then needs to be updated and managed

Reply:
changed it so that this lives on PanelContext